great_expectations/data_context/store/datasource_store.py

```diff
@@ -127,15 +149,19 @@ def delete(self, datasource_config: DatasourceConfig) -> None:
         self.remove_key(self._build_key_from_config(datasource_config))
 
     def _build_key_from_config(  # type: ignore[override]
-        self, datasource_config: DatasourceConfig
+        self, datasource_config: DatasourceConfig | FluentDatasource
     ) -> Union[GXCloudIdentifier, DataContextVariableKey]:
-        return self.store_backend.build_key(
-            name=datasource_config.name,
-            id=datasource_config.id,
-        )
+        if datasource_config.id:
+            id_: str | None = str(datasource_config.id)
+        else:
+            id_ = datasource_config.id  # type: ignore[assignment] # None or empty string
```

Comment:
Should the `id_` type declaration go above the conditional?

Reply:
Done